### PR TITLE
Adopt shared mutation-testing workflow (#571)

### PR DESCRIPTION
**Pull request by leynos**
## Summary

This branch replaces Wireframe's bespoke mutation-testing GitHub Actions implementation with a thin caller of the shared reusable workflow from `leynos/shared-actions`, pinned to the merge commit that introduced the shared workflow. It keeps the existing `Mutation testing` workflow name and cadence while moving detection, sharding, cargo-mutants installation, and summary generation into the shared workflow.

Closes #571.

Issue: [#571](https://github.com/leynos/wireframe/issues/571)
Roadmap task: none identified in branch evidence.
ExecPlan: [docs/execplans/adopt-shared-mutation-workflow.md](https://github.com/leynos/wireframe/blob/adopt-shared-mutation-workflow/docs/execplans/adopt-shared-mutation-workflow.md)

The branch implements the ExecPlan's pre-merge stages: it adopts the shared workflow, enables `--all-features` for mutation runs, excludes example and test-support scaffolding from mutation targets, updates ADR-007 to record the superseded bespoke implementation, and refreshes the developers' guide. The remaining Stage E acceptance check is post-merge runtime validation through a manual workflow dispatch.

## Review walkthrough

- Start with [.github/workflows/mutation-testing.yml](https://github.com/leynos/wireframe/blob/adopt-shared-mutation-workflow/.github/workflows/mutation-testing.yml) to review the new caller: triggers, explicit permissions, ref-scoped concurrency, the pinned shared workflow SHA, `wireframe_testing`, `exclude-globs`, and `--all-features`.
- Then review [docs/execplans/adopt-shared-mutation-workflow.md](https://github.com/leynos/wireframe/blob/adopt-shared-mutation-workflow/docs/execplans/adopt-shared-mutation-workflow.md) for the implementation rationale, constraints, validation model, and post-merge acceptance work.
- Review [docs/adr-007-mutation-testing-with-cargo-mutants.md](https://github.com/leynos/wireframe/blob/adopt-shared-mutation-workflow/docs/adr-007-mutation-testing-with-cargo-mutants.md) for the amended decision record explaining that the bespoke implementation has been superseded while the decision remains accepted.
- Finish with [docs/developers-guide.md](https://github.com/leynos/wireframe/blob/adopt-shared-mutation-workflow/docs/developers-guide.md) to see the contributor-facing mutation-testing guidance after the workflow move.

## Validation

- `git diff --check origin/main...HEAD`: passed with no output.
- `python3 -c "import yaml; yaml.safe_load(open('.github/workflows/mutation-testing.yml'))"`: passed with no output.
- `actionlint .github/workflows/mutation-testing.yml`: passed with no output.
- `uvx zizmor .github/workflows/mutation-testing.yml`: passed; no findings reported.
- `make markdownlint`: passed; `Summary: 0 error(s)`.
- `make nixie`: passed; all diagrams validated successfully.
- `coderabbit review --agent`: passed; `findings: 0`.

## Notes

The branch intentionally does not change Rust source. The first full behavioural acceptance signal remains post-merge: dispatch `Mutation testing` manually and confirm shard fan-out, merged reporting, and removal of the issue #571 false-survivor classes.

## Summary by Sourcery

Adopt the shared mutation-testing reusable GitHub Actions workflow and document the new caller-based setup, while keeping existing schedule and workflow naming.

CI:
- Replace the bespoke mutation-testing GitHub Actions workflow with a thin caller of the shared reusable workflow, including ref-based concurrency, minimal permissions, and configuration for extra crate dirs, excluded paths, and --all-features.

Documentation:
- Update the developers guide to describe the shared mutation-testing workflow, sharded execution, and configuration choices, and amend ADR-007 to record that the bespoke implementation was superseded by the shared workflow while the decision remains accepted.
- Add an execution plan document detailing the rationale, constraints, risks, and validation steps for adopting the shared mutation-testing workflow.

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai resume

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already. (Treat documentation and validation coverage as in scope).

### ❌ Failed checks (1 error, 2 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing (Overall) | ❌ Error | No automated tests or contract checks cover the workflow change; only static linting/docs validation and a post-merge manual dispatch are present. | Add an automated workflow contract test or integration check that exercises the caller’s inputs, ref handling, and concurrency, rather than relying on manual dispatch. |
| User-Facing Documentation | ⚠️ Warning | The PR changes mutation-testing behaviour, but docs/users-guide.md has no mutation-testing coverage and is unchanged in the diff. | Add a users-guide section for the shared workflow, --all-features, excluded paths, and the manual-dispatch sharded run model. |
| Testing (Unit And Behavioural) | ⚠️ Warning | The branch adds no unit or behavioural tests; the execplan states no workflow test harness exists and only static validation plus a post-merge manual dispatch are planned. | Add an automated workflow test or integration harness that exercises the caller boundary; do not rely on linting and a post-merge manual dispatch alone. |

File: tests/workflow_contract.rs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    );
+    assert!(
+        WORKFLOW.contains("contents: read"),
+        "the calling job must grant contents: read"
+    );
+    assert!(
+        WORKFLOW.contains("id-token: write"),
+        "the calling job must grant id-token: write for OIDC source resolution"
+    );
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/workflow_contract.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by pandalump** [comment]
@coderabbitai review

**Comment by pandalump** [comment]
@coderabbitai review
